test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -0,0 +1,208 @@
+        self.assertTrue(os.path.isdir(log_path))
+        # Check that 'mom_logs' were captured
+        log_path = os.path.join(output_dir, "node", "mom_logs")
+        self.assertTrue(os.path.isdir(log_path))
+        # Check that 'comm_logs' were captured
+        log_path = os.path.join(output_dir, "comm", "comm_logs")
+        self.assertTrue(os.path.isdir(log_path))
+        # Check that 'sched_logs' were captured
+        log_path = os.path.join(output_dir, "scheduler", "sched_logs")
+        self.assertTrue(os.path.isdir(log_path))
+        # Check that 'accounting_logs' were captured
+        log_path = os.path.join(
+            output_dir, "server", "server_priv", "accounting")
+        self.assertTrue(os.path.isdir(log_path))
+
+        # Cleanup
+        self.du.rm(path=output_dir, recursive=True, force=True)
+
+    def test_snapshot_basic(self):
+        """
+        Test capturing a snapshot via the pbs_snapshot program
+        """
+        output_dir = "mysnapshot_basic"
+        snap_cmd = ["pbs_snapshot", "-o", output_dir]
+        ret = self.du.run_cmd(cmd=snap_cmd)
+        self.assertTrue(ret['rc'] == 0)
+
+        # Cleanup
+        self.du.rm(path=output_dir, recursive=True, force=True)
+
+    def test_snapshot_without_logs(self):
+        """
+        Test capturing a snapshot via the pbs_snapshot program
+        Capture no logs
+        """
+        output_dir = "mysnapshot_nologs"
+        snap_cmd = ["pbs_snapshot", "-o", output_dir, "--service-logs=0",
+                    "--accounting-logs=0"]
+        ret = self.du.run_cmd(cmd=snap_cmd)
+        self.assertTrue(ret['rc'] == 0)
+
+        # Check that 'server_logs' were not captured
+        log_path = os.path.join(output_dir, "server", "server_logs")
+        self.assertTrue(os.path.isdir(log_path) is False)
+        # Check that 'mom_logs' were not captured
+        log_path = os.path.join(output_dir, "node", "mom_logs")
+        self.assertTrue(os.path.isdir(log_path) is False)
+        # Check that 'comm_logs' were not captured
+        log_path = os.path.join(output_dir, "comm", "comm_logs")
+        self.assertTrue(os.path.isdir(log_path) is False)
+        # Check that 'sched_logs' were not captured
+        log_path = os.path.join(output_dir, "scheduler", "sched_logs")
+        self.assertTrue(os.path.isdir(log_path) is False)
+        # Check that 'accounting_logs' were not captured
+        log_path = os.path.join(output_dir, "server", "server_priv",
+                                "accounting")
+        self.assertTrue(os.path.isdir(log_path) is False)
+
+        # Cleanup
+        self.du.rm(path=output_dir, recursive=True, force=True)
+
+    def test_snapshot_version(self):
+        """
+        Test the --version option to pbs_snapshot
+        """
+        snap_cmd = ["pbs_snapshot", "--version"]
+        ret = self.du.run_cmd(cmd=snap_cmd)
+        self.assertTrue(ret['rc'] == 0)
+        print ret['out']
```

Comment:
I would skip this test.  It's too hard to test if the --version option is correct.  This only tests that it returns 0.  That isn't much of a test.